### PR TITLE
S3: s3 access point reads / writes reach destination bucket

### PR DESCRIPTION
**Pull request by rafcio19**
Resolves https://github.com/getmoto/moto/issues/6912

### CODE REVIEW

**Comment by rafcio19** [comment]
@bblommers do the test fails look genuine? If not, could you rerun them?

**Comment by bblommers** [comment]
Restarted @rafcio19. There are still three TF tests failing, but I can't see/understand why - they would need to be run locally to see what's wrong with them.

**Comment by bblommers** [comment]
Actually @rafcio19, other PR's also have failing TF tests with the same generic error (`fatal error: concurrent map writes`), so I'm guessing it's not related to these changes.

File: moto/s3/models.py [hunk 1]
```diff
@@ -1762,6 +1762,20 @@ def get_bucket(self, bucket_name: str) -> FakeBucket:
             account_id = s3_backends.bucket_accounts[bucket_name]
             return s3_backends[account_id]["global"].get_bucket(bucket_name)
 
+        # when access point is used the url that's passed is of following format:
+        # AccessPointName-AccountId.s3-accesspoint.*Region*.amazonaws.com
+        # only the first part is passed as bucket name: AccessPointName-AccountId
+        if bucket_name.endswith(f"-{self.account_id}"):
```
[the remark this answers is not in the document]
Comment (rafcio19) [reply moto/s3/models.py]: I had a look but there is no special header in the request to indicate access point call; `host` header it is.

**Comment by rafcio19** [comment]
Something funky happening with those TF test 🤨

**Comment by bblommers** [comment]
> Something funky happening with those TF test 🤨

Def - I'll try to fix them separately in #6928

**Review by bblommers — APPROVED** [review]
LGTM - thanks @rafcio19!

**Pull request merged**